### PR TITLE
[BB-370] Non staff users now get 403 error when trying to request aggregator data about other users

### PR DESCRIPTION
**Pull request by Metfriet**
**Description:** Changed the behavior to give a 403 error when a non-staff users tries to request data about all/another user. Updated the tests to reflect this change. 

**JIRA:** https://tasks.opencraft.com/browse/BB-370

**Dependencies:** -
**Merge deadline:** -
**Installation instructions:** -

**Testing instructions:**

Use something like postman to call the API.

**Reviewers:**
- None selected yet

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:**  Do we need to add a Changelog for this?

### CODE REVIEW

**Comment by lgp171188** [comment]
@Metfriet, you have to update the docstring for the views using the `CompletionViewMixin` to specify the behaviour corresponding to the changes introduced in this PR.

File: completion_aggregator/api/common.py [hunk 1]
```diff
@@ -120,8 +120,11 @@ def user(self):
 
         requested_username = self.request.GET.get('username')
         if not requested_username:
-            user = self.request.user
-            self._requested_user = None
+            if self.request.user.is_staff:
+                user = self.request.user
+                self._requested_user = None
+            else:
+                raise PermissionDenied()
```
Comment (lgp171188) [inline completion_aggregator/api/common.py]: @Metfriet, you have to add unit tests for this scenario where the user is trying to access the endpoint without providing a username query string parameter.

File: completion_aggregator/api/v1/views.py [hunk 2]
```diff
@@ -68,8 +68,8 @@ class CompletionListView(CompletionViewMixin, APIView):
 
         username (optional):
             The username of the specified user for whom the course data is
-            being accessed.  If not specified, this defaults to the requesting
-            user.
+            being accessed.  If not specified, this will show data for all users
+            if performed by a staff user, otherwise it will throw a 403 Error.
```
[the remark this answers is not in the document]
Comment (Metfriet) [reply completion_aggregator/api/v1/views.py]: @lgp171188 Yes I think requested would be better.

File: completion_aggregator/api/v1/views.py [hunk 3]
```diff
@@ -258,10 +258,11 @@ class CompletionDetailView(CompletionViewMixin, APIView):
 
         username (optional):
             The username of the specified user for whom the course data is being
-            accessed.
+            accessed. If a non-staff users tries to access another users data they
```
[the remark this answers is not in the document]
Comment (Metfriet) [reply completion_aggregator/api/v1/views.py]: @lgp171188 Yes it should, thanks for pointing that out.

File: completion_aggregator/api/common.py (hunk 1 above)
Comment (Metfriet) [reply completion_aggregator/api/common.py]: @lgp171188 I've added 2 unit tests for this.

File: completion_aggregator/api/v1/views.py [hunk 4]
```diff
@@ -258,10 +258,11 @@ class CompletionDetailView(CompletionViewMixin, APIView):
 
         username (optional):
             The username of the specified user for whom the course data is being
-            accessed.
+            accessed. If a non-staff users try to access another user's data they
```
Comment (lgp171188) [inline completion_aggregator/api/v1/views.py]: @Metfriet, nit: remove `a` since the subject is in plural.

Comment (Metfriet) [reply completion_aggregator/api/v1/views.py]: @lgp171188 fixed, sorry for the hassle

**Comment by lgp171188** [comment]
@Metfriet, unless there is a specific reason not to, we usually squash the commits of a PR before merging it (see the penultimate item in the merge checklist). I see that @jcdyer has taken care of creating a new tag, publishing it to PyPI. So the post-merge checklist is taken care of.

**Comment by Metfriet** [comment]
@lgp171188 sorry about that. Is it possible to still squash commits after the branch has been merged already?

**Pull request merged**